src/RestServerGateway.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+namespace Omnipay\SagePay;
+
+use Omnipay\SagePay\Message\ServerRestCompletePurchaseRequest;
+use Omnipay\SagePay\Message\ServerRestMerchantSessionKeyRequest;
+use Omnipay\SagePay\Message\ServerRestPurchaseRequest;
+use Omnipay\SagePay\Message\ServerRestRefundRequest;
+use Omnipay\SagePay\Message\ServerRestRepeatRequest;
+use Omnipay\SagePay\Message\ServerRestRetrieveTransactionRequest;
+use Omnipay\SagePay\Message\ServerRestVoidRequest;
+
+/**
+ * Sage Pay Rest Server Gateway
+ */
+class RestServerGateway extends ServerGateway
+{
+    public function getName()
+    {
+        return 'Sage Pay REST Server';
+    }
+    
+    public function getUsername()
+    {
+        return $this->getParameter('username');
+    }
+
+    public function setUsername($value)
+    {
+        return $this->setParameter('username', $value);
+    }
+
+    public function getPassword()
+    {
+        return $this->getParameter('password');
+    }
+
+    public function setPassword($value)
+    {
+        return $this->setParameter('password', $value);
+    }
+
+    /**
+     * Create merchant session key (MSK).
+     */
+    public function createMerchantSessionKey(array $parameters = array())
+    {
+        return $this->createRequest(ServerRestMerchantSessionKeyRequest::class, $parameters);
+    }
+
+    /**
+     * Purchase and handling of return from 3D Secure redirection.
+     */
+    public function purchase(array $parameters = array())
+    {
+        return $this->createRequest(ServerRestPurchaseRequest::class, $parameters);
+    }
+
+    /**
+     * Handle purchase notifcation callback.
+     */
+    public function complete(array $parameters = array())
+    {
+        return $this->createRequest(ServerRestCompletePurchaseRequest::class, $parameters);
+    }
+
+    /**
+     * Get transaction information from Sage.
+     */
+    public function getTransaction(array $parameters = array())
```

Comment:
Could this be renamed to `fetchTransaction`? That would bring it in line with the optional methods listed in `Omnipay\Common\GatewayInterface` and hopefully improve consistency with other gateways following that interface.

Reply:
Sure, I don't see why not. Happy to make any changes that would bring it further in line with how it should be